internal/framework/status/updater.go

```diff
@@ -154,16 +153,29 @@ func (upd *UpdaterImpl) updateNginxGateway(ctx context.Context, status *NginxGat
 	upd.cfg.Logger.Info("Updating Nginx Gateway status")
 
 	if status != nil {
-		upd.writeStatuses(ctx, status.NsName, &ngfAPI.NginxGateway{}, func(object client.Object) {
-			ng := object.(*ngfAPI.NginxGateway)
-			ng.Status = ngfAPI.NginxGatewayStatus{
-				Conditions: convertConditions(
-					status.Conditions,
-					status.ObservedGeneration,
-					upd.cfg.Clock.Now(),
-				),
-			}
-		})
+		statusUpdatedCh := make(chan struct{})
+
+		go func() {
+			upd.writeStatuses(ctx, status.NsName, &ngfAPI.NginxGateway{}, func(object client.Object) {
+				ng := object.(*ngfAPI.NginxGateway)
+				ng.Status = ngfAPI.NginxGatewayStatus{
+					Conditions: convertConditions(
+						status.Conditions,
+						status.ObservedGeneration,
+						upd.cfg.Clock.Now(),
+					),
+				}
+			})
+			statusUpdatedCh <- struct{}{}
```

Comment:
I think it is possible for the goroutine to block trying to write to the `statusUpdatedCh` here.

If the context is canceled during the execution of `writeStatuses` and the `updateNginxGateway` function exits before `writeStatuses` returns, then there will be no receiver for this channel and it will block on write. 

Since `writeStatuses` will return if the context is canceled, I don't think we need the goroutine in this function. 

We can just call `writeStatuses` normally in the function body.

Aplologies for not catching this when we were rebasing the PR on my changes.